src/scripting.md

```diff
@@ -5,66 +5,179 @@ eleventyNavigation:
 
-Blank lines and lines beginning with // are ignored so you can embed comments in a script.
-Script commands that operate on a DOM element identify the DOM element with a format of attribute=value where the attribute identifies a unique attribute of the DOM element that you want to act on. For example, if you are filling out a form and the element you want to populate looks like this:
+A script lets you automate a WebPageTest test. Here’s how it looks:
 
-```html
-<input type="text" class="tabInputFields" id="lgnId1" value="" tabindex="1" maxlength="97" name="loginId"/>
+```markup
+// EXAMPLE: log into a site before measuring a page
+
+// Disable logging: we don’t want the login page to appear in test results
+logData 0
+
+// Visit the login page, enter the username and the password into the fields, and press “Submit”
+navigate https://my-site.com/login
+setValue name=email darth@vader.com
+setValue name=password foooooooooooooooorce
+click innerText=Sign in
+
+// Now, as we’ve signed in, enable logging again
+logData 1
+
+// And go the page we’re measuring
+navigate https://my-site.com/dashboard
+```
+
+Here’s another example:
+
+```markup
+// EXAMPLE: measure what happens when we interact with the page
+
+// Go the page we’re measuring
+navigate https://my-site.com/dashboard
+
+// Give it a bit of time to initialize after loading
+sleep 5
+
+// Open the support dialog and wait for the page to become idle
+execAndWait document.querySelector('.chat-dialog > button').click()
 ```
 
-you could identify it as **id=lgnId1** , **name=loginId** or **tabindex=1**
+See more [commands](#recommended-commands) and [examples](#sample-scripts) below.
+
+## Recommended Commands
+
+For historical reasons, if a script step fails, it does so silently. If something doesn’t work, unfortunately, you won’t get any feedback why. If you’re new to scripting, we recommend using the following commands only. It’s easier to debug them as you can run them manually yourself.
+
+#### `navigate`
+Navigates the browser to the provided URL and waits for it to complete.
+Browser Support: IE, Chrome, Firefox, Safari
+
+```markup
+usage:   navigate <url>
+example: navigate http://webmail.aol.com
+
+<url> - URL to provide the browser for navigation (same as you would enter into the address bar)
+```
+
+#### `exec`
+Executes JavaScript.
+Browser Support: IE, Chrome, Firefox
+```markup
+usage:   exec <JavaScript code>
+example: exec window.setInterval('window.scrollBy(0,600)', 1000);
+```
+
+#### `execAndWait`
+Executes JavaScript and waits for the browser to complete any activity generated from the action.
+Browser Support: IE, Chrome, Firefox
+```markup
+usage:   execAndWait <JavaScript code>
+example: execAndWait window.setInterval('window.scrollBy(0,600)', 1000);
+```
+
+## Full Reference
+
+### Comments
+
+A line that starts with  `// ` is ignored. Use this for comments.
+
+```markup
+// Example comment
+navigate not-a-comment.com
+```
+
+### Selectors
```

Comment:
I had no idea `innerText=` exists – until I edited this doc today. To make this easier to discover, I

* put selectors into a separate section
* linked to this section from every command that works with selectors.